src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1992,28 +1990,74 @@ static bool updateContextAttributeItem
   // given that the logic in NGSIv2 for specifying location attributes is gogint to change
   // (the best moment to address this FIXME is probably once NGSIv1 has been deprecated and
   // removed from code)
-  if (targetAttr->getLocation().length() > 0 && targetAttr->name != locAttr)
```

Comment:
The location stuff in this function has been heavily re-written. Hard to follow in diff mode. I'd recommend to have a look to the final code of this part directly.